### ISSUE TITLE
[➕ Feature]: support secrets in workflows

### ISSUE THREAD

**Issue by talboren** [keephq/keep#1712]
As a user, instead of keeping raw variables (which may be secrets) in my workflows, I would like to be able to reference a secret using " {{ secrets.SECRET_NAME }}".

PRD:
1. New screen in the settings page, that allows one to CRUD tenant secrets.
2. New API that utilizes `SecretManagerFactory.get_secret_manager` to CRUD secrets in the backend
3. Load relevant workflow secrets to `context_manager` to allow reference in a workflow

**Comment by sameerkali** [comment]
Hi @talboren  , I'm interested. Can I help fix this?

**Comment by talboren** [comment]
@sameerkali absolutely, you can give it a try! happy to help more on our [Slack](https://slack.keephq.dev)
